tests/v2/test_waku_rln_relay_onchain.nim

```diff
@@ -278,14 +278,13 @@ procSuite "Waku-rln-relay":
     let pk2 = keyPair2.idCommitment.toUInt256()
     debug "membership commitment key", pk2 = pk2
 
-    var events = [newFuture[void](), newFuture[void]()]
-    proc handler(pubkey: Uint256, index: Uint256): RlnRelayResult[void] =
-      debug "handler is called", pubkey = pubkey, index = index
-      if pubkey == pk:
-        events[0].complete()
-      if pubkey == pk2:
-        events[1].complete()
-      let isSuccessful = rlnPeer.rlnInstance.insertMember(pubkey.toIDCommitment())
+    let event = newFuture[void]()
+    var handler: GroupUpdateHandler
+    handler = proc (blockNumber: BlockNumber, 
+                    members: seq[MembershipTuple]): RlnRelayResult[void] =
+      debug "handler is called", members = members
+      event.complete()
+      let isSuccessful = rlnPeer.rlnInstance.insertMembers(0, members.mapIt(it.idComm))
       check:
         isSuccessful
```

Comment:
If possible, I'd move all the checks to the end of the test case, under the "THEN" section. Checks within a `Future` usually are problematic when debugging.

This can be taken as a separate piece. Not blocking, but important.

Reply:
Tracking https://github.com/status-im/nwaku/issues/1358 thanks!